apps/andi/test/integration/andi_web/live/access_group_live_view/edit_access_group_live_view_test.exs

```diff
@@ -353,16 +343,114 @@ defmodule AndiWeb.AccessGroupLiveView.EditAccessGroupLiveViewTest do
       html = render_click(save_button)
 
       eventually(fn ->
-        access_group = AccessGroups.get(access_group.id) |> Andi.Repo.preload(:datasets) |> IO.inspect(label: "here")
+        access_group = AccessGroups.get(access_group.id) |> Andi.Repo.preload(:datasets)
         assert [%Dataset{id: ^dataset_id}] = access_group.datasets
       end)
     end
   end
 
+  test "removes a dataset when remove action is clicked", %{curator_conn: conn} do
```

Comment:
Is there a way to test concurrent actions in Elixir reliably? I'm not opposed to writing the test, though I'm not sure how I'd do it since simultaneous action is hard to simulate without introducing unrealistic flakiness. I can think of two situations where this could happen. One would be if on the manage datasets modal someone selected and removed so fast that the events could be processed out of order, but I don't see any reason logically that would be an issue. The other would be if two data curators were modifying the same access group at the same time in contradictory ways. What existing protections do we have against concurrent modification?